pkg/trace/providers.go

```diff
@@ -0,0 +1,68 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/getoutreach/gobox/internal/call"
+	"github.com/getoutreach/gobox/pkg/log"
+	"github.com/getoutreach/gobox/pkg/orerr"
+	"github.com/getoutreach/gobox/pkg/statuscodes"
+)
+
+// provider is a generic interface for tracking calls.
+type provider interface {
```

Comment:
This is only a subset of the interface proposed but felt like it was safer to start with the smallest step rather than going overboard here.